### ISSUE TITLE
[Inquiry/Feature Request] Coverage Difference on Pull Requests

### ISSUE THREAD

**Issue by YaleChen299** [MishaKav/pytest-coverage-comment#185]
As per title, it would be great if the comment can display the coverage difference of the branches of the PR.

Something like this:

coverage: $\color{green} 94.3\%$. Improved by $\color{green} +2.1\%$ when pulling from `feat` to `main` .
coverage: $\color{green} 90.0\%$. Remain the same$ when pulling from `feat` to `main` .
coverage: $\color{green} 89.3\%$. Dropped by $\color{red} -1.1\%$ when pulling from `feat` to `main`.

Let me know if this is already a feature! Thanks for the work! Enjoyed this workflow!

**Comment by aviralpostman** [comment]
I had made some changes in my fork to enable this. If this is what you require, I can try raising a PR:

<img width="834" alt="Image" src="https://github.com/user-attachments/assets/db1badc1-9fed-400b-a577-4da3488c86c6" />

To compare the coverage difference, I'm storing the coverage info in cache i.e. whenever a PR is merged, a workflow will get triggered on the `main` branch, run the tests there and update the coverage information in the `gh-actions` cache. The action would thus look require the `push` event to be part of the workflow.

**Comment by ruthus18** [comment]
@aviralpostman looks great! Could you please upload your fork to public or raise the PR?

**Comment by CuriousLearner** [comment]
Hi @aviralpostman Can you please link to your fork?

**Comment by Alex-ley-scrub** [comment]
If you click on @aviralpostman's profile you will see https://github.com/aviralpostman/jest-coverage-comment-aviral-fork which looks like is the fork in question (but for jest not pytest, although I'm sure the idea is the same)

**Comment by aviralpostman** [comment]
yes, it's for jest. I had implemented a working solution for my org and wanted to contribute to this repo via the fork. But, something is not working in my fork 🥺. The idea is store the coverage in gh-actions cache and pull it to compare. 
I'll see if I can wrap the implementation this week. But, if someone is interested to pick up, he/she is most welcome

**Comment by mschoettle** [comment]
This would be a great enhancement and another thing that allows to fully migrate from external services like codecov.

**Comment by mschoettle** [comment]
I noticed today that GitHub now has a feature for this which can then be enabled in a branch ruleset to "Restrict code coverage". Unfortunately, it requires the "code quality" feature which is a paid feature (on top of a plan as far as I understand).

Here are the docs:

- https://docs.github.com/en/code-security/how-tos/maintain-quality-code/set-up-code-coverage
- https://docs.github.com/en/code-security/how-tos/maintain-quality-code/restrict-code-coverage

This is the `upload-code-coverage` action: https://github.com/actions/upload-code-coverage

Maybe this can be somewhat replicated as follows:

- on the default branch: this action could upload coverage data (overwrite previous one if it exists)
- on a PR: this action compares the coverage data to the one from the default branch
- on a PR: the comparison is used to enhance the comment with the coverage diff to the default branch
